csrc/ir/nodes.cpp

```diff
@@ -2823,12 +2823,16 @@ IterDomain* IterDomain::resize(
   if (iter_type_opt.has_value()) {
     iter_type = iter_type_opt.value();
   } else if (left_expansion->isConstInt() && right_expansion->isConstInt()) {
-    if (resized_id_size->isConstInt()) {
+    auto left = left_expansion->evaluateInt();
+    auto right = right_expansion->evaluateInt();
+    if (left == 0 && right == 0) {
+      // Trivial Resize
+      return in;
```

Comment:
It seems odd to check the trivial case here as this part is to figure out the iteration type. Why not just do this check early? Also, maybe we should check the optional iteration type is not given or the same as the input iteration type.

Reply:
Good point. I moved the check to the beginning of the function and added an assertion that the specified `IterType` does not clash, if it is given.